File: PAT/javaSrc/PAT-B1014.福尔摩斯的约会.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String[] week = { "MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN" };
        String str1 = in.nextLine();
        String str2 = in.nextLine();
        String str3 = in.nextLine();
        String str4 = in.nextLine();
        boolean isFindWeek = false;
        boolean isFindHour = false;
        for (int i = 0; i < str1.length() && i < str2.length(); i++) {
            char ch1 = str1.charAt(i);
            char ch2 = str2.charAt(i);
            if (ch1 == ch2) {
                if (!isFindWeek && ch1 >= 'A' && ch1 <= 'G') {
                    System.out.print(week[ch1 - 65] + " ");
                    isFindWeek = true;
                    continue;
                }
                if (isFindWeek && !isFindHour) {
                    if (ch1 >= '0' && ch1 <= '9') {
                        System.out.printf("%02d:", (ch1 - 48));
                        isFindHour = true;
                    }else if (ch1 >= 'A' && ch1 <= 'N') {
                        System.out.print((ch1 - 55) + ":");
                        isFindHour = true;
                    }
                }
            }
        }
        for (int i = 0; i < str3.length() && i < str4.length(); i++) {
            char ch3 = str3.charAt(i);
            char ch4 = str4.charAt(i);
            if (ch3 == ch4) {
                if (ch3 >= 'A' && ch3 <= 'Z' || ch3 >= 'a' && ch3 <= 'z') {
                    System.out.printf("%02d", i);
                }
            }
        }
    }
}

File: PAT/cSrc/PAT-B1014.福尔摩斯的约会.cpp
#include <iostream>
#include <cctype>
using namespace std;
int main()
{
    string a, b, c, d; /* 输入四个非空字符串 */
    cin >> a >> b >> c >> d;
    char t[2]; /* 记录前两个字符串中相同元素 */
    int pos, i = 0, j = 0;
    while (i < a.length() && i < b.length()) /* 防止 i 越界 */
    {
        /* 如果字符串 a 中的元素和字符串 b 中的元素相匹配，并且在 A~G 范围之内*/
        if (a[i] == b[i] && (a[i] >= 'A' && a[i] <= 'G'))
        {
            t[0] = a[i]; /* 星期 */
            break;
        }
        i++; /* i 后移 */
    }
    i = i + 1; /* 结束第一轮匹配之后，后移 */
    while (i < a.length() && i < b.length())
    {
        /* isdigit 函数：用于检查其参数是否为十进制数字字符 */
        if (a[i] == b[i] && ((a[i] >= 'A' && a[i] <= 'N') || isdigit(a[i])))
        {
            t[1] = a[i]; /* 钟头 */
            break;
        }
        i++;
    }
    while (j < c.length() && j < d.length()) /* 防止 j 越界 */
    {
        /* isalpha 函数：如果参数是字母字符，函数返回非零值，否则返回零值 */
        if (c[j] == d[j] && isalpha(c[j]))
        {
            pos = j; /* 计算分钟 */
            break;
        }
        j++; /* j 后移 */
    }
    string week[7] = {"MON ", "TUE ", "WED ", "THU ", "FRI ", "SAT ", "SUN "}; /* 某星期的 3 字符缩写 */
    int m = isdigit(t[1]) ? t[1] - '0' : t[1] - 'A' + 10;                      /* 计算钟头 */
    cout << week[t[0] - 'A'];                                                  /* 输出星期数 */
    printf("%02d:%02d", m, pos);                                               /* 输出时间 */
    return 0;
}
